compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/cfg/ControlFlowGraphBuilder.kt

```diff
@@ -436,27 +436,34 @@ class ControlFlowGraphBuilder {
     }
 
     fun exitClass(): Pair<ClassExitNode?, ControlFlowGraph>? {
+        assert(currentGraph.kind == ControlFlowGraph.Kind.Class)
         if (currentGraph.declaration == null) {
-            graphs.pop().also { assert(it.kind == ControlFlowGraph.Kind.Class) }
+            graphs.pop()
             return null
         }
 
         // Members of a class can be visited in any order, so data flow between them is unordered,
         // and we have to recreate the control flow after the fact.
-        assert(currentGraph.kind == ControlFlowGraph.Kind.Class)
-        val klass = currentGraph.declaration as FirClass
+        val enterNode = lastNodes.pop() as ClassEnterNode
+        val exitNode = currentGraph.exitNode as ClassExitNode
+        val klass = enterNode.fir
+        if ((klass as FirControlFlowGraphOwner).controlFlowGraphReference != null) {
+            // TODO: IDE LL API sometimes attempts to analyze a enum class while already analyzing it, causing
+            //  this graph to be built twice (or more). Not sure what this means. Nothing good, probably.
+            //  In any case, attempting to add more edges to subgraphs will be fatal.
+            graphs.pop()
+            return null
```

Comment:
This isn't the first time I'm encountering weirdness around enum entries, does it really make sense for them to be represented as anonymous objects? Shouldn't they be more like named objects?

Reply:
Enum entry does not have its own type. Type of `SomeEnum.A` is always `SomeEnum`, not `A`. So that's why they were made an anonymous objecs